### DISCUSSION TITLE
Remove triggered dataset event for specific dags

### DISCUSSION THREAD

**Discussion by Zoynels** [apache/airflow#35682] [Q&A]
Is there any way to remove some of the dataset triggers for specific dag? For example, there are 2 datasets already updated. But I need to remove one or all of them.
As I understand, this information is located in the airflow database in table dataset_dag_run_queue and I can manually remove them from the database. But for this, I need access to the database. How can I remove them from UI?

![image](https://github.com/apache/airflow/assets/5841616/8ecc39bb-daae-4b71-901a-ecfae42d2515)

**Comment by Zoynels** [comment]
For now I have second way to remove from airflow specific dag, but it also remove data from database:
```python
from airflow.utils.db import provide_session
from airflow.models.dataset import DatasetDagRunQueue
with DAG(
    dag_id="datasets_use_7_remove_dataset_dag_run_queue_entry",
    schedule_interval=None,
    start_date=pendulum.datetime(2023, 1, 1),
    tags=["datasets_use"],
) as dag7:

    @provide_session
    def remove_dataset_dag_run_queue_entry(target_dag_id, session=None):
        session.query(DatasetDagRunQueue).filter(
            DatasetDagRunQueue.target_dag_id == target_dag_id
        ).delete()


    @task(provide_context=True)
    def remove_file7(**context):
        target_dag_id = "datasets_use_6_read_account_dimension__fact_table1__fact_table2"
        remove_dataset_dag_run_queue_entry(target_dag_id=target_dag_id)

    remove_file7()
```

**Reply by Amin-Siddique** [reply]
Nice Solution! Any idea how can this be generalized?
